coordinator/gscoordinator/utils.py

```diff
@@ -830,67 +830,141 @@ def _pre_process_for_output_graph_op(op, op_result_pool, key_to_op, **kwargs):
     )
 
 
-def _pre_process_for_project_to_simple_op(op, op_result_pool, key_to_op, **kwargs):
+def _pre_process_for_project_to_simple_op(  # noqa: C901
+    op, op_result_pool, key_to_op, **kwargs
+):
     # for nx graph
     if op.attr[types_pb2.GRAPH_TYPE].graph_type in (
         graph_def_pb2.DYNAMIC_PROJECTED,
         graph_def_pb2.ARROW_FLATTENED,
     ):
         return
-    assert len(op.parents) == 1
+
+    def __check_v_prop_exists_in_all_v_labels(schema, prop):
+        exists = True
+        for v_label in schema.vertex_labels:
+            exists = exists and schema.vertex_property_exists(v_label, prop)
+        return exists
+
+    def __check_e_prop_exists_in_all_e_labels(schema, prop):
+        exists = True
+        for e_label in schema.edge_labels:
+            exists = exists and schema.edge_property_exists(e_label, prop)
+        return exists
+
     # get parent graph schema
+    assert len(op.parents) == 1
```

Comment:
Could we raise a `ValueError` or `RuntimeError` instead?

Reply:
`Assert` failed means there are some internal bugs in our DAG implementation since the number of parents in this case is always 1.